### PR TITLE
Proj0032: Migrate away from BinaryFormatter

### PR DESCRIPTION
**Pull request by Corniel**
Documentation for https://github.com/dotnet-project-file-analyzers/dotnet-project-file-analyzers/issues/253

### CODE REVIEW

File: rules/Proj0032.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Proj0032: Migrate away from BinaryFormatter
+Starting with .NET 9, Microsoft no longer includes an implementation of
+`BinaryFormatter` in the runtime (it has [been depricated](https://learn.microsoft.com/en-us/dotnet/core/compatibility/serialization/5.0/binaryformatter-serialization-obsolete)
```
Comment (sungam3r) [inline rules/Proj0032.md:3]:
```suggestion
`BinaryFormatter` in the runtime (it has [been deprecated](https://learn.microsoft.com/en-us/dotnet/core/compatibility/serialization/5.0/binaryformatter-serialization-obsolete)
```

File: rules/Proj0032.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+with the release of .NET 5). The APIs are still present, but their
+implementation always throws an exception, regardless of project type.
+
+Setting the existing backwards compatibility flag is no longer sufficient
+to use BinaryFormatter with .NET 9, as is explained in [this blog post](https://devblogs.microsoft.com/dotnet/binaryformatter-removed-from-dotnet-9/).
```
Comment (sungam3r) [inline rules/Proj0032.md:8]:
```suggestion
to use `BinaryFormatter` with .NET 9, as is explained in [this blog post](https://devblogs.microsoft.com/dotnet/binaryformatter-removed-from-dotnet-9/).
```

File: rules/Proj0032.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+Due to the fact that Microsoft discourages the usage as strongly as they
+do, this rule reports violations for all target frameworks, and
```
Comment (sungam3r) [inline rules/Proj0032.md:11]:
```suggestion
can, this rule reports violations for all target frameworks, and
```

?

File: rules/Proj0032.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+also when used in combination with [System.Runtime.Serialization.Formatters"](https://learn.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-migration-guide/compatibility-package).
```
Comment (sungam3r) [inline rules/Proj0032.md:12]:
```suggestion
also when used in combination with [System.Runtime.Serialization.Formatters](https://learn.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-migration-guide/compatibility-package).
```

File: rules/Proj0032.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+can, this rule reports violations for all target frameworks, and
+also when used in combination with [System.Runtime.Serialization.Formatters](https://learn.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-migration-guide/compatibility-package).
+
+## Non-compliant
+``` XML
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net9.0</TargetFramework>
+	<EnableUnsafeBinaryFormatterSerialization>true</EnableUnsafeBinaryFormatterSerialization>
```
Comment (CptWesley) [inline rules/Proj0032.md:20]: indentation doesn't match

**Review by CptWesley — APPROVED** [review]
1 comment, otherwise LGTM

**Pull request merged**